.estafette.yaml

```diff
@@ -10,16 +10,29 @@ version:
   semver:
     major: 1
     minor: 2
+    patch: 1
+    labelTemplate: 'beta-{{auto}}'
+    releaseBranch: 1.2.1
```

Comment:
So version will be changed manually from now on?

Reply:
Yes, that's the idea. You update it to the next milestone when you start working on it.

When you're done with all the stories for the next milestone and want to release it you create and push a branch equal to `releaseBranch` and it will build the final release docker container and Helm package.

The benefit is that we can do not just major and minor releases, but also patch releases. And the version numbers for chart and app version are equal and better looking.

Reply:
Sounds good.
Maybe we can add this process to README file.

Reply:
Yes certainly. Right now I'm still comparing git workflows so we can choose one and stick to it. Once I've figured that out I'll add it to the readme. See https://www.atlassian.com/git/tutorials/comparing-workflows for available options. I'm wondering if we should pick `gitflow` because it creates release branches or a hybrid of `feature branch workflow` since it still merges everything back to master. Do you have experience with _gitflow_?

Reply:
I worked with the full blown gitflow workflow before, for a software that needed manual testing. It was quite complex with feature, releases and hotfix branches.
Personally I find the gitlab flow a little less complex. For the case of estafette projects I think the release branches model would fit, see https://docs.gitlab.com/ee/topics/gitlab_flow.html#release-branches-with-gitlab-flow .
But it still requires cherry-picking, which adds some cognitive overhead.

Reply:
It seems more suitable since we're unlikely to very frequently need to hotfix anything and we do get the benefit of the release branches, but not the burden of the development branch.